Sources/ApolloCodegenLib/Frontend/JavaScript/src/index.ts

```diff
@@ -69,16 +71,31 @@ export function mergeDocuments(documents: DocumentNode[]): DocumentNode {
   return concatAST(documents);
 }
 
+function buildSchemaExtensions(userExtensions?: readonly [DocumentNode]): DocumentNode {
+  let documents = [apolloCodegenSchemaExtension];
+  if (userExtensions) {
+    documents = documents.concat(userExtensions);
+  }
+  return mergeDocuments(documents)
+}
+
+function mergeSchemaExtensions(extensions: DocumentNode, schema: GraphQLSchema): GraphQLSchema {
+  return extendSchema(schema, extensions)
+}
+
 export function validateDocument(
   schema: GraphQLSchema,
   document: DocumentNode
 ): readonly GraphQLError[] {
-  return validate(schema, document, defaultValidationRules);
+  const codegenSchema = mergeSchemaExtensions(buildSchemaExtensions(), schema)
+  return validate(codegenSchema, document, defaultValidationRules);
 }
 
 export function compileDocument(
   schema: GraphQLSchema,
   document: DocumentNode
 ): CompilationResult {
-  return compileToIR(schema, document);
+  console.log(schema.getDirectives());
```

Comment:
Was this `log` command meant to stick around?

Reply:
Nope, thought I caught that. Thanks